packages/controllers/src/services/PluginExecutionEnvironmentService.ts

```diff
@@ -0,0 +1,25 @@
+import { JsonRpcRequest } from 'json-rpc-engine';
+import { PluginData } from '@mm-snap/types';
+
+export interface PluginMetadata {
+  hostname: string;
+}
+
+export type TerminatePlugin = (pluginName: string) => void;
+export type Command = (
+  pluginName: string,
+  message: JsonRpcRequest<unknown>,
+) => Promise<unknown>;
+export type TerminateAll = () => void;
+export type CreatePluginEnvironment = (
+  metadata: PluginMetadata,
+) => Promise<string>;
+export type StartPlugin = (pluginData: PluginData) => Promise<unknown>;
+
+export interface PluginExecutionEnvironmentService {
+  terminatePlugin: TerminatePlugin;
+  terminateAllPlugins: TerminateAll;
+  createPluginEnvironment: CreatePluginEnvironment;
+  command: Command;
```

Comment:
In line with my other comments, I think these make sense on this interface, but I they shouldn't be used by the plugin controller directly, if possible.